src/SqlAsyncCollector.cs

```diff
@@ -237,14 +239,32 @@ private async Task UpsertRowsAsync(IEnumerable<T> rows, SqlAttribute attribute,
         /// to upsert to and returns the extra property names in a List.
         /// </summary>
         /// <param name="columns"> The columns of the table to upsert to </param>
+        /// <param name="rowItem"> Sample row used to get the column names when item is a JObject </param>
         /// <returns>List of property names that don't exist in the table</returns>
-        private static IEnumerable<string> GetExtraProperties(IDictionary<string, string> columns)
+        private static IEnumerable<string> GetExtraProperties(IDictionary<string, string> columns, T rowItem)
         {
+            if (typeof(T) == typeof(JObject))
+            {
+                Dictionary<string, string> dictObj = (rowItem as JObject).ToObject<Dictionary<string, string>>();
+                return dictObj.Keys.Where(prop => !columns.ContainsKey(prop))
+                .Select(prop => prop);
+            }
             return typeof(T).GetProperties().ToList()
                 .Where(prop => !columns.ContainsKey(prop.Name))
                 .Select(prop => prop.Name);
         }
 
+        private static IEnumerable<string> GetColumnNamesFromPOCO(T row, bool bracketed = false)
```

Comment:
Rename to `GetColumnNamesFromItem`

And please add comments explaining the reasoning for getting columns in this way for each type